lib/validator.js

```diff
@@ -17,37 +23,65 @@ var inMapping = {
  * @param parameters list of params
  * @returns JSON schema
  */
-function convertToJsonSchema(parameters) {
+Validator.prototype._convertToJsonSchema = function(parameters) {
     var schema = {
         type: 'object',
         properties: {}
     };
 
     parameters.forEach(function(param) {
-        if (!schema.properties[inMapping[param.in]]) {
-            schema.properties[inMapping[param.in]] = {
-                type: 'object',
-                properties: {}
-            };
-            // 'required' array must have at least one element according to json-schema spec,
-            // se we can't preinitialize it.
-            schema.required = schema.required || [];
-            schema.required.push(inMapping[param.in]);
-        }
-        var reqPartSchema = schema.properties[inMapping[param.in]];
-        reqPartSchema.properties[param.name] = {};
-        if (param.required) {
-            reqPartSchema.required = reqPartSchema.required || [];
-            reqPartSchema.required.push(param.name);
+        if (param.in !== 'body') {
+            if (!schema.properties[inMapping[param.in]]) {
+                schema.properties[inMapping[param.in]] = {
+                    type: 'object',
+                    properties: {}
+                };
+                // 'required' array must have at least one element according to json-schema spec,
+                // se we can't preinitialize it.
+                schema.required = schema.required || [];
+                schema.required.push(inMapping[param.in]);
```

Comment:
@gwicke no, @d00rman is actually correct. If we have 2 different parameters in the request part (e.g. query), it will get to required multiple times. I'll fix that.